packages/flame/lib/src/game/flame_game.dart

```diff
@@ -86,6 +86,24 @@ class FlameGame extends ComponentTreeRoot
     setMounted();
   }
 
+  /// Setting to true will remove all children and process any remaining life
+  /// cycle events.
+  bool disposeAllOnRemove = false;
+
+  /// Called when the game is about to be removed from the Flutter widget tree,
+  /// but before it is actually removed.
+  ///
+  /// You can override this for additional behavior, but for
+  ///  [disposeAllOnRemove] to be triggered, you must call super.
+  @override
+  void onRemove() {
+    if (disposeAllOnRemove) {
+      removeAll(children);
```

Comment:
Those will already be ignored, a component can't be removed twice without being re-added again.
Me and Munsterlander is going to have a chat about the current implementation, so it might come to change soon.